task/02-numbers-tasks.js

```diff
@@ -221,7 +250,14 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    let number = parseFloat(value);
+    if (number) {
+        return number;
+    } else if (number === 0) {
+        return number;
+    } else {
+        return def;
+    }
```

Comment:
JS has method isNaN, please, use it in solution
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN